packages/web-components/fast-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,324 @@
+                    break;
+            }
+        } else {
+            switch (keyCode) {
+                case keyCodeArrowUp:
+                    event.preventDefault();
+                    this.decrement(tabsLength);
+                    break;
+                case keyCodeArrowDown:
+                    event.preventDefault();
+                    this.increment(tabsLength);
+                    break;
+            }
+        }
+        switch (keyCode) {
+            case keyCodeHome:
+                event.preventDefault();
+                this.activeTabIndex = 0;
+                this.setTabs();
+                break;
+            case keyCodeEnd:
+                event.preventDefault();
+                this.activeTabIndex = tabsLength - 1;
+                this.setTabs();
+                break;
+        }
+    };
+
+    private handleActiveIndicatorPosition() {
+        if (this.activeIndicator) {
+            if (this.ticking) {
+                this.activeIndicatorRef.style.transform = "translateX(0px)";
+                this.activeIndicatorRef.classList.remove("activeIndicatorTransition");
+                if (this.isHorizontal()) {
+                    this.activeIndicatorRef.style.gridColumn = `${
+                        this.activeTabIndex + 1
+                    }`;
+                } else {
+                    this.activeIndicatorRef.style.gridRow = `${this.activeTabIndex + 1}`;
+                }
+                this.ticking = false;
+            } else {
+                this.ticking = true;
+
+                this.animateActiveIndicator();
+            }
+        }
+    }
+
+    private animateActiveIndicator(): void {
+        if (this.isHorizontal()) {
+            const prev: number = this.activeIndicatorRef.offsetLeft;
+            this.activeIndicatorRef.style.gridColumn = `${this.activeTabIndex + 1}`;
+            const next: number = this.activeIndicatorRef.offsetLeft;
+            this.activeIndicatorRef.style.gridColumn = `${this.prevActiveTabIndex + 1}`;
+            const dif: number = next - prev;
+            this.activeIndicatorRef.style.transform = `translateX(${dif}px)`;
+            this.activeIndicatorRef.classList.add("activeIndicatorTransition");
+            this.activeIndicatorRef.addEventListener("transitionend", () => {
+                this.ticking = false;
+                this.activeIndicatorRef.style.gridColumn = `${this.activeTabIndex + 1}`;
+                this.activeIndicatorRef.style.transform = "translateX(0px)";
+                this.activeIndicatorRef.classList.remove("activeIndicatorTransition");
+            });
+        } else {
+            const prev: number = this.activeIndicatorRef.offsetTop;
+            this.activeIndicatorRef.style.gridRow = `${this.activeTabIndex + 1}`;
+            const next: number = this.activeIndicatorRef.offsetTop;
+            this.activeIndicatorRef.style.gridRow = `${this.prevActiveTabIndex + 1}`;
```

Comment:
It seems like this logic is the same, it's just the properties that you're using that are different. Maybe instead of duplicating the logic you could conditionalize the properties into variables, and then use those variables in property accessors